custom_components/fordpass/sensor.py

```diff
@@ -171,10 +171,16 @@ def get_value(self, ftype):
                     return { "Ambient Temp": ambient_temp}
                 return None
             if self.sensor == "fuel":
-                if "fuelRange" in self.data:
-                    return {"fuelRange" : self.units.length(self.data.get("fuelRange", {}).get("value", 0),UnitOfLength.KILOMETERS)}
-                if "xevBatteryRange" in self.data:
-                    return {"batteryRange": self.units.length(self.data.get("xevBatteryRange", {}).get("value", 0),UnitOfLength.KILOMETERS)}
+                fuel = {}
+                fuel_range = self.data.get("fuelRange", {}).get("value", 0)
+                battery_range = self.data.get("xevBatteryRange", {}).get("value", 0)
+                if fuel_range != 0:
+                    # Display fuel range for both Gas and Hybrid (assuming its not 0)
+                    fuel["fuelRange"] = self.units.length(fuel_range,UnitOfLength.KILOMETERS)
+                if battery_range != 0:
+                    # Display Battery range for EV and Hybrid
+                    fuel["batteryRange"] = self.units.length(battery_range,UnitOfLength.KILOMETERS)
+                return fuel
```

Comment:
Was hoping this rework will show:

GAS:
fuel level - and fuel range attribute

HYBRID:
fuel level - fuel range attribute
              - battery range attribute
EV:
battery level - battery range attribute